From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Christian=20K=C3=B6nig?= <christian.koenig@amd.com>
Date: Tue, 15 Jun 2021 15:15:01 +0200
Subject: dma-buf: use new iterator in dma_resv_copy_fences

This makes the function much simpler since the complex
retry logic is now handled else where.

Signed-off-by: Christian König <christian.koenig@amd.com>
Reviewed-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Link: https://patchwork.freedesktop.org/patch/msgid/20211005113742.1101-5-christian.koenig@amd.com

(Backported from commit 96601e8a4755d333a8d2e826134d5220ae2e8c24 linux)

Reviewed-by: Arvind Yadav <arvind.yadav@intel.com>
Signed-off-by: Varun Gupta <varun.gupta@intel.com>
---
 drivers/dma-buf/dma-resv.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/dma-buf/dma-resv.c b/drivers/dma-buf/dma-resv.c
index 5563cd80ded8..5b9c09dff9f5 100644
--- a/drivers/dma-buf/dma-resv.c
+++ b/drivers/dma-buf/dma-resv.c
@@ -418,10 +418,10 @@ int dma_resv_copy_fences(struct dma_resv *dst, struct dma_resv *src)
 
 		dst_list->shared_count = 0;
 		for (i = 0; i < src_list->shared_count; ++i) {
-			struct dma_fence __rcu **dst;
+			enum dma_resv_usage usage;
 			struct dma_fence *fence;
 
-			fence = rcu_dereference(src_list->shared[i]);
+			dma_resv_list_entry(src_list, i, NULL, &fence, &usage);
 			if (test_bit(DMA_FENCE_FLAG_SIGNALED_BIT,
 				     &fence->flags))
 				continue;
@@ -437,8 +437,8 @@ int dma_resv_copy_fences(struct dma_resv *dst, struct dma_resv *src)
 				continue;
 			}
 
-			dst = &dst_list->shared[dst_list->shared_count++];
-			rcu_assign_pointer(*dst, fence);
+			dma_resv_list_set(dst_list, dst_list->shared_count++,
+					  fence, usage);
 		}
 	} else {
 		dst_list = NULL;
-- 
2.43.0

